--- ./syntax/pgsql2.vim	2019-04-26 15:38:36.149195827 +0200
+++ ./syntax/pgsql.vim	2019-04-26 15:38:53.629268014 +0200
@@ -1907,6 +1907,7 @@
 
 syn region plpgsql matchgroup=sqlString start=+\$pgsql\$+ end=+\$pgsql\$+ keepend contains=ALL
 syn region plpgsql matchgroup=sqlString start=+\$body\$+ end=+\$body\$+ keepend contains=ALL
+syn region plpgsql matchgroup=sqlString start=+\(AS\|DO\)\s\+\$\$+ end=+\$\$+ keepend contains=ALL
 if get(g:, 'pgsql_dollar_strings', 0)
   syn region sqlString start=+\$\$+ end=+\$\$+ contains=@Spell
 else
